### PR TITLE
Added a new FallbackToClientPathTransformer that will use the UrlPathTransformer and fallback to the client if neccesary

### PR DESCRIPTION
**Pull request by digeff**
For some programs the debugger might not have enough information to figure out how to map all the URLs to file paths, so we can ask the client if he knows how to do it

@roblourens @rakatyal

### CODE REVIEW

File: src/chrome/chromeDebugAdapter.ts [hunk 1]
```diff
@@ -220,6 +221,10 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
             });
         }
 
+        if (args.supportsMappingURLsToFilePaths) {
+            this._pathTransformer = new FallbackToClientPathTransformer(this._session);
```
Comment (roblourens) [inline src/chrome/chromeDebugAdapter.ts]: Do this at the top of the method so we are less likely to add some code working on pathTransformer before this.

File: src/transformers/fallbackToClientPathTransformer.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                        resolve(err ? false : true);
+                    });
+                } catch (e) {
+                    resolve(false);
+                }
+            }).then(doesFilePathExist => doesFilePathExist
+                // If it does, we use that result
+                ? filePath
+                // If it doesn't, we ask the client to map it as a fallback
+                : this.requestClientToMapURLToFilePath(scriptUrl).catch(rejection => {
```
Comment (roblourens) [inline src/transformers/fallbackToClientPathTransformer.ts]: Can include eval scripts which are given a fake url like `VM45`. Not sure whether you want to send those back to the client.

Comment (digeff) [reply src/transformers/fallbackToClientPathTransformer.ts]: That is a good question. I'll consult with a PM.

File: src/chrome/chromeUtils.ts [hunk 3]
```diff
@@ -281,3 +281,9 @@ export function selectBreakpointLocation(lineNumber: number, columnNumber: numbe
 
     return locations[0];
 }
+
+export var EVAL_NAME_PREFIX = 'VM';
```
Comment (roblourens) [inline src/chrome/chromeUtils.ts]: const

**Pull request merged**